src/client/db.rs

```diff
@@ -0,0 +1,109 @@
+use bson;
+use bson::Bson;
+use client::MongoClient;
+use client::coll::Collection;
+use client::common::{ReadPreference, WriteConcern};
+
+/// Interfaces with a MongoDB database.
+pub struct Database<'a> {
+    pub name: String,
+    pub client: &'a MongoClient,
+    read_preference: Option<ReadPreference>,
```

Comment:
The [CRUD spec](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#q-a) indicates that ```read_preference``` and ```write_concern``` should be specified on a Client/Database/Collection level, and both the python and c++ drivers follow this paradigm with the client also having these fields. The python driver also removed the ability to specify these at the operations level, while the c++ driver provides these as options for certain operations.